### PR TITLE
Code base cleanup o clock

### PR DESCRIPTION
**Pull request by ChrisRu**
This turned out bigger than I had hoped and I'm sorry 

### Chore:
- Improved performance of ESLint and TypeScript by ignoring more files.

### General:
- Added better data fetching error handling in a few places. Now shows when fetching goes wrong, instead of just a permanent spinner.
- Wrapped some component side-effects in `useEffect`, so it's only called once and not on every rerender.

### Schedule:
- Fix scrolling to specific date (now without hash-links)
- Modified the Virtuoso grouping and data fetching logic to be a bit simpler

### Bookmarking:
- Moved bookmark data to context, because Virtuoso wasn't having a good time with the effects accessing local storage in all the `ScheduleCard`s.
- Disable bookmarking runs with no ID specified, this was breaking the bookmarks as well.

### Themes
- Fix app breaking for events where no theme is set in Horaro.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -63,53 +72,104 @@ const Themes = {
...
+        await cancelNotification(bookmark.notificationId);
+      }
+    } else {
+      const {notifications} = await scheduleNotification({
+        title: 'Your run is about to start!',
+        body: `In about an hour ${run.game} - ${run.category} starts`,
+        scheduled: dayjs(run.scheduled).subtract(1, 'hour').toDate(),
+      });
+
+      StoreBookmark({run, notificationId: notifications[0].id});
```
Comment (olidotjpeg) [inline src/App.tsx]: Curious, why are these functions named with capital letters?

File: src/components/common/EstimateParser.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ interface IProps {
 
 function EstimateParser({seconds}: IProps) {
   const parsedTimer = new window.Date(seconds * 1000).toISOString().substr(11, 8);
-  return <React.Fragment>{parsedTimer}</React.Fragment>;
+  return <>{parsedTimer}</>;
```
Comment (olidotjpeg) [inline src/components/common/EstimateParser.tsx]: Can we please continue using `React.Fragment` here? I prefer that syntax way more :D

File: src/pages/BookmarksPage.tsx [hunk 3]
```diff
@@ -35,9 +31,18 @@ function BookmarkPage({location}: RouteComponentProps) {
       </StyledHeaderWrapper>
       <Content>
         <ScheduleList>
-          {bookmarks.map(({run}) => (
-            <ScheduleCard key={run.scheduled + (run.players.join('-') || '')} run={run} />
-          ))}
+          {bookmarks.size === 0 ? (
+            <p>You don't have any bookmarks yet. Go to the schedule and favorite some!</p>
```
Comment (olidotjpeg) [inline src/pages/BookmarksPage.tsx]: `You don't have any bookmarks yet. Go to the schedule and mark some!`
Maybe this wording is better?

File: src/App.tsx (hunk 1 above)
Comment (ChrisRu) [reply src/App.tsx]: Idk, was like that before. I too prefer lowercase for non components.

File: src/components/common/EstimateParser.tsx (hunk 2 above)
Comment (ChrisRu) [reply src/components/common/EstimateParser.tsx]: Ah okay, I like the `<>` syntax, but sure no worries!

**Review by olidotjpeg — APPROVED** [review]

**Pull request merged**